### ISSUE TITLE
(DUPLICATE) Moving from Slack to another service

### ISSUE THREAD

**Issue by benigeri** [reactiflux/volunteers#17]
# PLEASE DISCUSS IN https://github.com/reactiflux/volunteers/issues/25
## Why don't we leave Slack?

In short, we haven't found a better alternative. 
#### 1. High switching cost

7500 people already have Reactiflux Slack account. Dozens of repos link to their respective Reactiflux channel. Switching to another service would be a big deal, and we'll probably lose a significant chunk of the community. 

We're not opposed to moving away from Slack. We have very high switching cost though, so we'll only move if we have very good reasons to do so. 
#### 2. The product is amazing

Slack is great, and lots of people love it. Many of us believe it's the best messaging service for teams. 
#### 3. Many of us have it open by default.

Reactiflux is unusually active compared to other public Slack groups, Gitter channels, IRC channels, etc… Why is that? 

Probably because so many of us use React **and** Slack at work. 

Many of us are thinking about React at work. We’re working using libraries, building libraries, and running into bugs and problems. So we’re inclined to talk (read complain) about it with other people. 

Many of us work at startups and tech companies that use Slack for team communication. So we open Slack every day and leave it running almost all the time. 

And several months ago, Slack made it incredibly to switch between all of your teams. 

Which means that even though most of us could never justify having yet another app or webpage open 24/7, just to talk to Reactiflux people, it happens automatically. 

Obviously this isn’t the case for everyone on Reactiflux, but it is for many of us. In the early days, tons of people that joined kept on telling me “Oh, it’s so nice to be able to ask React questions on Slack since I already have it open all day.” 
### [Gitter](https://gitter.im/)

Gitter's benefits are that: 
1. It works well with GitHub, which probably everybody on Reactiflux users. 
2. It comes with unlimited message history.

But Reactiflux would not make sense in Gitter world. Gitter channels are based on repos. So instead of what we currently have, in Slack world: 
- reactiflux/react-router
- reactiflux/redux
- reactiflux/fluxible

Gitter world would look like this: 
- rakt/react-router
- facebook/react
- yahoo/fluxible

All channels would no longer be grouped under Reactiflux. The community would break up and separate into dozens of smaller organizations, like Facebook, Yahoo, and Rakt. (Are we even sure that Facebook or Yahoo want Gitter organizations?) 

So in Gitter world, what would be left of Reactiflux? Not much. Why do we need another Reactiflux/redux room? There will already be a Rakt/redux room. 

I don't think Gitter world would be better. 

In Slack world, Reactiflux creates an overlap between all of the React sub-communities. It brings people from #redux, #alt, #reflux, and #fluxible together because it's so easy to go from one channel to the other, or to mingle in #general. 

In Slack world, it's easy to find channels and to peek inside to see if you like it. You can tag people across channels, so everybody's around all the time. 

Slack let's Reactiflux be a higher order community for all of the smaller React and Flux communities.

**Comment by eldh** [comment]
It won't solve points 1 (though it seems user accounts can be migrated) or 3, and I'm not sure about 2 either. But anyway, I would like to throw in http://www.mattermost.org/ as an alternative. It's built with react, it's open source, and at least on paper it provides a similar feature set as Slack. The operating cost would be limited to the cost of running a server. It's at least better fit to the Reactiflux community model than Gitter.

**Comment by zackify** [comment]
I have used mattermost and let me assure you: it is far too buggy to be
used yet. I wish it was stable, but it has a lot of problems. Bugs. Tons of
them. No native apps, and a lot more.

**Comment by eldh** [comment]
Ok. Too bad, it seems like an interesting project and it would be awesome if there was a viable open platform alternative.

**Comment by benigeri** [comment]
Yup, I bet that it will get some love one day and become a viable option for many communities. But today it feels like a premature product. Lots of bugs and missing features. It's really hard to beat Slack's product quality and polish.

**Comment by aaronjensen** [comment]
fwiw, you can have rooms within a gitter, so you could do: reactiflux/f/redux, reactiflux/f/alt (where f is just a short name I picked out for a project).

If projects wanted their own room, that'd be fine, you could just have reactiflux be a curated list of react gitter rooms, many of which are hosted at reactiflux/f/*

**Comment by vcarl** [comment]
> Sam C (Slack)
> Sep 24, 3:24 PM
> 
> Hi Paul,
> 
> I’m writing to inform you that we've disabled the ability to add more users to the Reactiflux Slack team. We're happy that you've found Slack to be a great platform for your community but from both an administrative and performance perspective this is proving to be unsustainable.
> 
> Although we do simple chat and file-sharing very well, Slack simply isn’t designed for communities of thousands of users to chat. Slack's ideally designed for teams of coworkers who collaborate closely and frequently to get work done. That said, looking into ways to better support communities like yours in the future is something that has been suggested many times! The idea is under consideration, and if changes are implemented to make this easier in the future, we'll be sure to get the word out.
> 
> I’m not sure how you’re sending out your invites, because if you’re using the web interface it should tell you that your maximum user limit was reached. If you’re using the undocumented API, we’re returning a user_limit_reached error that you'll run into soon.
> 
> If you do want to continue to use Slack to manage this community, I'd recommend you spin up multiple smaller teams and cap each one around no more than 1,000 users. Once one team fills up, stop inviting people to that team and start a new one for the next group of people who want to join your community. That's the best way to ensure that your teams remain manageable and that the service stays nice and snappy for you.
> 
> Thanks so much for your understanding. If you have any questions, please don't hesitate to drop us a line!
> 
> Regards,
> 
> Sam

TL;DR Slack says Reactiflux is too big and needs to find a new home.

**Comment by robotmayo** [comment]
My heart belongs to IRC.

**Comment by cobbweb** [comment]
Another option: https://www.zulip.org/

**Comment by jaylaw81** [comment]
And https://github.com/RocketChat/Rocket.Chat

**Comment by vcarl** [comment]
We've talked about Rocket, Zulip, MatterMost, IRC, Gitter, etc. The main downside of a lot of these is that they're self-hosted, so Reactiflux would have to cover server costs and have somebody keeping it running. Given that Slack has mentioned performance and administration issues, that might not be viable. 
- Gitter: Few people _like_ it, but it's not terrible and they've supported communities of >7k in the past. It's hosted, so we don't need to worry about keeping it running, and it's free.
- IRC: Would work for the scale, but clients are meh and discoverability isn't as great.
- Rocket: Self-hosted, but otherwise looks good.
- Zulip: Looks like it would meet our needs, but it's still self-hosted.
- MatterMost: None of the people who'd used it had good things to say, and it's self-hosted.

**Comment by morenoh149** [comment]
http://www.reactiflux.com/ should be disabled then no? I've shared the link with colleagues and they have received no email. A note would suffice.

**Comment by Aaike** [comment]
i have the same problem , tried entering my email several times at reactiflux.com, i get no invitation email...
how can i get an invite ?

**Comment by germ13** [comment]
@morenoh149 
I believe the invite screen is rendered by [slackin](https://github.com/rauchg/slackin).

The `README.md` there states one reason why emails wont be received.
Might be a good idea to look into this.  

Although, there are a multiple places where things might be going wrong:
- slackin itself and/or config
- server config
- mail server (i dont know what is handling this)

**Comment by morenoh149** [comment]
@Aaike @germ13 no. We found out in this thread that slack is not allowing new members to join reactiflux. I'm suggesting the signup page be disabled altogether or replaced with a note.

**Comment by vcarl** [comment]
Agreed. @benigeri, I'd be happy to do some of that, but I'm unaware of where anything is hosted or where the code lives.

**Comment by vcarl** [comment]
>Hey Paul,
>
>Thanks for getting back to me. We generally don't offer support over the phone but I brought this up to our leadership team to get some clarity on where we stand.
>
>In it's current form, Slack just isn't optimized for large teams. As previously mentioned, we'd really want to be able to support large open source teams in the long run.
>
>There are many reasons why Slack doesn't work well for large communities. A lot of the pain points that these large community teams bring up to us are not going to be addressed in the near future. This is due to our limited resources which is currently focused on enhancing and optimizing Slack for the purpose it was initially designed for.
>
>Although many communities are willing to deal with these limitations, this isn't the experience that we want our users to have. We're hoping people will find services that are better designed for their needs as a whole rather than deal with the growing disparities of how Slack is currently designed and how teams are wanting to use it.
>
>Again, we'd think the best option would be to split the community into multiple smaller teams or you could consider having a team consisting of only proactive users and then relaying the information to the greater community elsewhere. I know this may not be ideal and may even be counter intuitive for a community, but I'm afraid we really don't have a better solution at the moment.
>
>If you'd still like a phone call, I'd be happy to coordinate with someone from the team although I'm not sure if we'll have anything more to add. Let me know your thoughts!    
>
>Sam

**Comment by BerkeleyTrue** [comment]
Hello Everybody,

I work for Free Code Camp. A couple of months ago we also ran into the same issue that the Reactiflux community ran into with Slacks invisible user limit. Unlike this community, we where given no warning that we where going to hit the ceiling. I won't go into further detail, but you can read about it [here](http://blog.freecodecamp.com/2015/06/so-yeah-we-tried-slack-and-we-deeply-regretted-it.html) if you have a couple of minutes. 

We switched to Gitter. We had an exceedingly warm welcome from the team and assurances that they could accommodate our community as it grows. In our main chat room we have over 24 and a half thousand users and we have [500 rooms](https://gitter.im/orgs/FreeCodeCamp/rooms) under Free Code Camps organization. There has not been any noticeable drop in quality of service. In fact we have noticed a steady climb. 

The Gitter team is incredibly accessible. We know them on a first name basis (The team is only four strong). Can you say that about Slack? And even though they are located in England we have have met with the team in San Francisco twice now. They let us in on what their plans are and how they could effect us in the future. They give us sneak peaks into new unannounced features (I'm extremely excited about one of them and can't wait for their public announcement).

Reactiflux is going to be huge and ever expanding. I think Gitter is the perfect platform. Just the response we got from the Gitter team when they heard Slack decided to stop letting our new users join our chat is enough of a reason to switch.

**Comment by vcarl** [comment]
We actually saw your blog post about the difficulties you had, Gitter's pretty high on our list because of your positive experiences. It's cool to hear they're so accessible, that's definitely a plus.

**Comment by Aaike** [comment]
@BerkeleyTrue that is good to know , we are also using Gitter on the Aurelia team , Our user base isn't as big as yours but we have a little over 2000 members now , and have experienced this steady increase in quality like you mentioned.

Would be really great if the switch to Gitter could happen sooner rather than later since a lot of people including me are feeling a bit left out when it comes to Reactiflux ;)

**Comment by BerkeleyTrue** [comment]
Awesome. I'm really hoping every thing works out regardless of the platform y'all choose. 

When you all do switch one thing you could do is create a bot that responds to all messages in Slack that the community has moved and how to join the new platform. That will ensure people move or they will constantly see an annoying message for every one of theirs. One thing you definitely don't want is for people to hang around the old Slack.

**Comment by benigeri** [comment]
Unfortunately, we have to leave Slack. 

I got on the phone with a couple of Slack people last week, and they have decided not to support infinitely large public communities. Slack wants to focus on building team communication software, and groups like Reactiflux don't fit into that. 

The main issue is with our rapidly growing user base. We are already causing a decent amount of load on there servers, and many of us are already experiencing performance issues. It would take Slack time and effort to attend to these issues, and they have not allocated the resources to do that. 

The only way we could stay with Slack would be to cap our user base at < 8k. Probably around 1-3K. This obviously doesn't work for an open community. The most important thing about our community is that it's open and anyone can join in less than a minute. So when new people want to join the conversation, it's very low friction. We obviously can't keep the friction low if we start moderating accounts to keep our numbers low. 

This is a bit sad, but at least we know for sure that Slack is no longer the right platform for us. Now we can focus our efforts on finding a better platform for Reactiflux. 

I'm going to create a new issue to focus the discussion on which product/service we should migrate to. I hope we can make a group decision in the next 1-2 weeks.

**Comment by Sing-Li** [comment]
Please pardon me for this plug.    And feel free to delete this if you deem it inappropriate.

I'm from the [Rocket.Chat](https://Rocket.Chat/) project,  a 100% MIT-licensed [open sourced](https://github.com/rocketchat/Rocket.Chat) slack alternative.

We are in the midst of adding a provisioning and hosting service (attempting to do it free -  long term - for all true FOSS projects).   Please try us out here in our [support and demo community server](https://demo.rocket.chat/) - it is filled with helpful core devs and contributors around the clock during the work week.  

Contact any one of us.  We will gladly supply you with more information on the project/service - or assist with  migration whenever you are ready.

**Comment by chiedo** [comment]
I'm a fan of IRC. Let's keep things simple. IRC is easy if you want to participate in a conversation (via web options) It only gets difficult when you want to do more.

Rocket.Chat is looking pretty cool to me though. It's at least worth looking into.

**Comment by sgoldsmith** [comment]
Have you considered www.hipchat.com ? Native clients on all platforms (incl Linux), unlimited integrations + free for teams of all sizes forever. There's even an easy Slack importer to migrate all your current teams, channels & people over to the new instance. 

We have many customers operating at the scale you're looking for; give it a try!

**Comment by Aaike** [comment]
i know IRC is a popular option and i do like it.
but let's consider easily sharing screenshots, code/syntax highlighting, basic markdown formatting and searching chat history.
if i'm not wrong these are all things that are lacking in IRC

**Comment by commadelimited** [comment]
I'm indifferent as I'm not part of this community, but I recently heard about a free alternative to Slack called Ryver:
http://www.ryver.com/features/

**Comment by morenoh149** [comment]
I would love to see rocketChat get more communtiy support. But a gitter is fine too.

**Comment by elwayman02** [comment]
Hipchat is the way to go, almost as good as Slack

**Comment by juliankrispel** [comment]
Ah that sucks. It's like saying, oh we love open source and obviously use open source software in our product. We can't be asked to contribute a bit of money towards the community. Can't somebody sponsor this, like facebook or something? It'd really make sense and be good publicity!

**Comment by gajus** [comment]
One more for IRC.

**Comment by philholden** [comment]
Cannot use IRC at work. We use internally with our own server. But I get warnings if I connected to freenode. Some bots apparently use freenode to communicate.

**Comment by engelgabriel** [comment]
Guys, I can create a https://reactiflux.rocket.chat for your community if you want to give Rocket.Chat a try?

**Comment by mydigitalself** [comment]
We're talking soon, but I just wanted to point out:

> But Reactiflux would not make sense in Gitter world. Gitter channels are based on repos. So instead of what we currently have, in Slack world:
> 
> reactiflux/react-router
> reactiflux/redux
> reactiflux/fluxible
> Gitter world would look like this:
> 
> rakt/react-router
> facebook/react
> yahoo/fluxible

This isn't quite the case. We started out with 1:1 mapping with GitHub repos, but very quickly added the ability to create any channel as long as it has an owner within GitHub, so you can quite easily have:

reactiflux/react-router
reactiflux/redux
reactiflux/fluxible

Where those rooms are public channels owned by the reactiflux org.

We'd love to have you folks on Gitter, it's exactly what we built our product for and are entirely committed to developer communities and communication.

**Comment by quicksnap** [comment]
I'll add my vote toward any rich client with it's own collection of rooms. I think IRC would be a loss because of its lack of rich communication and higher bar of entry.

I thought Discord was _awesome_, but the lack of monospace formatting is a huge deal to me. Also, currently lacks an API, so we would lose our bots.

**Comment by jryans** [comment]
IRC seems just fine to me. Gitter has an IRC bridge, so I would probably use that if Gitter was chosen.

**Comment by elwayman02** [comment]
That is incorrect, Dan:
https://support.discordapp.com/hc/communities/public/questions/205313137-API-for-bots

**Comment by elwayman02** [comment]
https://github.com/Arcbot-Org/Arcbot

**Comment by quicksnap** [comment]
@elwayman02 Glad to be wrong!

**Comment by quicksnap** [comment]
I was pulling my info from here: https://dl.dropboxusercontent.com/spa/4moihw0kt47f2sy/bj3as-wh.png

**Comment by elwayman02** [comment]
Yea, it's not officially released/documented yet, but it's there.

**Comment by vcarl** [comment]
Discord is blocked by corporate webfilters as gaming related, so IMO it's not the best choice.

**Comment by elwayman02** [comment]
I think there's an argument to be made to get it whitelisted for work
purposes, though... I don't think it's a strong enough reason to discount
the best available product.

**Comment by elwayman02** [comment]
In fact, pretty much the only way Discord will _stop_ being blocked as
gaming related is if some non-gaming communities like Reactiflux start
using it. Otherwise, we create a vicious cycle where we don't use it
because it's blocked for being gaming, and thus the only people using it
_are_ gamers.

**Comment by vcarl** [comment]
It advertises itself as being for gamers, I don't think that's a concern for them. Their main userbase is people using it for gaming, and there are already features we won't be taking advantage of.

I think Gitter is the best option. Other than personal preference I haven't seen a reason it wouldn't work, and it's previously been extremely friendly to groups larger than ours in the past.

**Comment by taion** [comment]
FWIW the voice features on Discord are so cool that at least a few of us are thinking of trying it out for more work-related stuff.

**Comment by elwayman02** [comment]
Chat room discovery on Gitter is pretty awful. There doesn't seem to be a
way to create a channel list other than making people navigate to each
channel via the links/badges.

**Comment by quicksnap** [comment]
Do we have a sense of how many Reactiflux users would be affected by corporate blacklist?

**Comment by BerkeleyTrue** [comment]
> Chat room discovery on Gitter is pretty awful.

You would just navigate to <org>/rooms. See: https://gitter.im/orgs/FreeCodeCamp/rooms

**Comment by vcarl** [comment]
Also https://gitter.im/orgs/reactiflux/rooms, we made a few while exploring options a while ago

**Comment by mydigitalself** [comment]
@elwayman02 we're working on improving that, but it's certainly not the case that there isn't a channel list. in any room, there's a globe of the world in the top right corner and that will take you to the directory of rooms ala:

https://gitter.im/gitterHQ
http://gitter.im/dotnet

Admittedly it needs to be better surfaced, and we're working on some designs for that right now.

**Comment by chiedo** [comment]
Just my opinion but we've heard a lot of options at this point. Could someone authoritative narrow it down to a few top contenders now so the conversation can continue from there?

If I made the call, we would continue with the assumption that we'd now be choosing between one of the following:
IRC
Gitter
ChatGrape

Just my opinion though!

**Comment by vcarl** [comment]
The community has expressed interest in IRC and Discord, mainly. The group coordinators have settled on Gitter as the most viable option, but didn't discuss Discord at the time. Pretty much everything has an IRC bridge (and would be simple to bot), so I think the IRC crowd can remain content-ish no matter what platform. 

For the purposes of narrowing options, the shortlist is
- Gitter
- Discord

**Comment by chiedo** [comment]
So right now, we're practically deciding between the following?

Gitter
Discord
IRC

**Comment by vcarl** [comment]
IRC was decided to be nonviable because of the relatively high switching cost for people who don't already use IRC, and the availability of bridges for those who really want IRC. 

Also, lack of discoverability of rooms.

**Comment by engelgabriel** [comment]
@vcarl what about our offer to host Rocket.Chat for you guys?
I think it would be awesome opportunity for the community to use/support an nice open source project.

**Comment by taion** [comment]
IRC is also pointless because if you wanted to use Freenode#reactjs, you'd already be using it. There's no "switch to IRC option", the "switch to IRC option" is just "shut down Reactiflux and use Freenode#reactjs instead".

**Comment by chiedo** [comment]
Good points @vcarl and @taion.

So now it's just a matter of choosing between Gitter and Discord? 

I personally am in favor of trying Rocket.Chat and taking up the offer from @engelgabriel but if that's off the table, let's officially put it to rest

**Comment by geekgonecrazy** [comment]
Another contributor to Rocket.Chat here.  

Just something to think about.  We're rapidly growing, features are being added daily.  If you want feature X you can open a PR and get that feature in the core.  That's not something you can do with these other platforms.

/end plug

**Comment by Daniel15** [comment]
#25 seems to be the more relevant issue to discuss this in :)

**Comment by benigeri** [comment]
Thank you all for the great feedback. I've read it and will try to update https://github.com/reactiflux/volunteers/issues/25 to include the major points. 

Closing this issue to focus the conversation in https://github.com/reactiflux/volunteers/issues/25
